pkg/aggregator/time_sampler.go

```diff
@@ -195,6 +197,12 @@ func (s *TimeSampler) dedupSerieBySerieSignature(
 	}
 
 	for _, serie := range serieBySignature {
+		// it is the final stage before flushing the series to the serialisation
+		// part of the pipeline but also, here is a stage where all series have been
+		// generated & processed (even the ones generated from a histogram metric).
+		if blocklist.Test(serie.Name) {
```

Comment:
Does it need to check for nil blocklist here? You added a check in dogstatsd server code in the previous PR.

Reply:
Good catch, thank you.